sympy/polys/polytools.py

```diff
@@ -7938,3 +7938,134 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system_poly(polys, **kwargs):
+
+    if not polys:
+        return [[]]
+
+    factored_eqs = []
+    for poly in polys:
+        if not isinstance(poly, Poly):
+            raise TypeError("All inputs must be Poly instances")
+
+        try:
+            constant, factors = factor_list(poly, **kwargs)
+            if not factors and constant.is_zero:
+                return []
+            elif not factors:
+                continue
+            current_factors = []
+            if constant.is_number and constant != 0:
+                current_factors.extend((Poly(factor, poly.gens), []) for factor, _ in factors)
+            else:
+                current_factors.extend((Poly(factor, poly.gens), [Poly(constant, poly.gens)])
+                                       for factor, _ in factors)
+
+            if current_factors:
+                factored_eqs.append(current_factors)
+
+        except ComputationFailed:
+            if kwargs.get('modulus'):
+                raise NotImplementedError("multivariate polynomials over finite fields")
+            raise
+
+    if not factored_eqs:
+        return [[]]
+
+    return _combine_factor_systems_poly(factored_eqs)
+
+
+def _combine_factor_systems_poly(factored_eqs):
+
+    factor_dict = {}
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            key = factor
+            if key not in factor_dict:
+                factor_dict[key] = []
+            factor_dict[key].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    def is_minimal_valid_combination(combo):
+        combo_set = set(combo)
+        # Must satisfy all equations
+        if not all(any(factor in combo_set for factor in eq) for eq in eq_factors):
+            return False
+
+        for size in range(1, len(combo)):
+            for subset in combinations(combo, size):
+                if all(any(factor in set(subset) for factor in eq) for eq in eq_factors):
+                    return False
+        return True
+
+    result = []
+    all_factors = list(factor_dict.keys())
+
+    for size in range(1, len(all_factors) + 1):
+        for combo in combinations(all_factors, size):
+            if is_minimal_valid_combination(combo):
```

Comment:
There has to be a better way than this combinatoric loop.

Reply:
I wrote some code to eliminate the use of `combinations` but it's kind of a bulky block now, although I get why running combinatoric loops should be avoided. 

The code block now looks like:

```python
    def is_valid_combination(combo, eq_factors):
        """Check if combination satisfies all equations"""
        combo_set = set(combo)
        return all(any(factor in combo_set for factor in eq) for eq in eq_factors)

    def is_minimal(combo, eq_factors):
        """Check if combination is minimal"""
        combo_set = set(combo)
        for factor in combo:
            reduced = combo_set - {factor}
            if is_valid_combination(reduced, eq_factors):
                return False
        return True

    result = []
    used_combinations = set()

    def build_combination(current, start_idx):
        if is_valid_combination(current, eq_factors):
            if is_minimal(current, eq_factors):
                combo_key = frozenset(current)
                if combo_key not in used_combinations:
                    used_combinations.add(combo_key)
                    result.append([(factor, factor_dict[factor]) for factor in current])
            return

        for i in range(start_idx, len(list(factor_dict.keys()))):
            factor = list(factor_dict.keys())[i]
            if factor not in current:
                current.append(factor)
                build_combination(current, i + 1)
                current.pop()

    build_combination([], 0)
```

Reply:
I'm pretty sure that my original code handled this correctly and was much simpler than this.